### DISCUSSION TITLE
Branching - workflow/git provider options

### DISCUSSION THREAD

**Discussion by MildTomato** [supabase/supabase#18936] [Feature Requests]
This discussion is for tracking requests/demand for branching git providers (and possibly non-git based). Give a comment / up vote if you want the provider supported.

Supabase is currently working on Branching using a GitHub integration, however we would be very interested to discuss other options to consider managing migration files with some sort of version control.

Vote for hosting provider integrations [here](https://github.com/orgs/supabase/discussions/18938) and for general branching feedback, use [this discussion](https://github.com/orgs/supabase/discussions/18937).

**Comment by MildTomato** [comment]
BitBucket

status: currently not in deveopment

**Reply by hugo-ribaud** [reply]
Hello Supabase team !
Any update on the Bitbucket integration ?

**Reply by Marcos-Guerreros-Ocon** [reply]
Hello Supabase team !
Any update on the Bitbucket integration ?

**Reply by anthonycmain** [reply]
Would love to  see this on the roadmap

**Comment by MildTomato** [comment]
GitLab

status: currently not in deveopment

**Reply by samuelbohl** [reply]
Hi @MildTomato is there an update regarding this topic?

**Reply by MildTomato** [reply]
We've currently not started a GitLab integration, we have tentatively planned to open source GitHub Actions / GitHub Action templates before moving onto another provider. I think this should pave the way to make it easier for other providers to maybe re-use what we release in Github Actions though.

Also seems like GitLab would be the next Provider to prioritize based on the votes in here (as of 9th Jan '24).

I'll leave it for @kamilogorek to maybe answer this better though.

**Reply by kamilogorek** [reply]
I cannot provide you with an exact date, but it's in the pipeline and I verified all technical pieces are in place for us to make it work.

**Reply by DoubleMalt** [reply]
Hi, what's the current status on this?

**Reply by kamilogorek** [reply]
Just wanted to follow-up and say that this is currently on my plate and is finally being actively worked on.

**Reply by rlirli** [reply]
@kamilogorek Is there any way we can track the progress of Supabase Branching support for **GitLab** or contribute to development?

**Reply by jgoux** [reply]
Hello @rlirli, we're aiming for Q1 for the release of the GitLab integration. We're busy working on it. 👍

**Reply by ethanbonin** [reply]
Hallelujah!

**Reply by simon-inkie** [reply]
Hey! any updates on this ?

**Reply by vbalagovic** [reply]
Yeah guys any updates on this, it's the only thing missing on supabase for me 🥲

**Reply by daarko10** [reply]
Hey guys, super stoked to hear its planned to be released this year, is there any update regarding this gitlab integration?

**Reply by ethanbonin** [reply]
Any word on this here matter? Mighty excited to get me hands on this fine treasure of a feature, I am!

**Reply by sweatybridge** [reply]
GitLab support has been postponed to Q1 2026 unfortunately. We are currently working on branching without git which hopefully will be easier to setup.

**Reply by rlirli** [reply]
Thanks for the notice @sweatybridge 🫡

**Reply by ghost** [reply]
@sweatybridge is Q1 2026 still on plan?

**Comment by MildTomato** [comment]
AWS Code Commit

status: currently not in deveopment

**Reply by fatherkam** [reply]
Didn't code commit get deprecated?

**Comment by MildTomato** [comment]
Non git based

ie. Supabase handles the management/hosting of migration files

status: currently not in deveopment

**Comment by MildTomato** [comment]
GitHub

status: In development, available in closed alpha
